contracts/util/BatchWithdraw.sol

```diff
@@ -0,0 +1,23 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "../interfaces/ILockDealNFT.sol";
+
+contract BatchWithdraw {
+    ILockDealNFT public lockDealNFT;
+
+    constructor(ILockDealNFT _lockDealNFT) {
+        lockDealNFT = _lockDealNFT;
+    }
+
+    function batchWithdraw(uint256[] memory tokenIds) external {
+        bool isApproved = lockDealNFT.isApprovedForAll(msg.sender, address(this));
+        for (uint256 i = 0; i < tokenIds.length; i++) {
```

Comment:
if there are a lot of nfts, it's better to create a local variable for `tokenIds.length`